bb_server/src/plugin/types/particle.rs

```diff
@@ -0,0 +1,36 @@
+use panda::{
+  parse::token::Span,
+  runtime::RuntimeError,
+};
+use crate::{
+  particle,
+  particle::Particle,
+};
+use bb_server_macros::define_ty;
+use crate::plugin::types::util::PFPos;
+use std::str::FromStr;
+
+#[define_ty]
+impl PParticle {
+  info! {
+    wrap: Particle,
+
+    panda: {
+      path: "bamboo::particle::Particle",
+    },
+  }
+
+  pub fn new(name: &str, pos: &PFPos, force: bool, offset: &PFPos, count: u32, data: f32) -> Result<Self, RuntimeError> {
+    Ok(PParticle {
+      inner: Particle {
+        ty: particle::Type::from_str(name)
+          .map_err(|e| RuntimeError::Custom(e.to_string(), Span::call_site()))?,
+        pos: pos.inner,
+        long_distance: force,
+        offset: offset.inner,
+        count,
+        data,
+      },
+    })
+  }
+}
```

Comment:
`cargo fmt` is the way. make sure you're on the nightly toolchain, and it should just fix all this.